components/support/migration/src/main/java/mozilla/components/support/migration/FennecSettingsMigrator.kt

```diff
@@ -0,0 +1,100 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.support.migration
+
+import android.content.Context
+import android.content.Context.MODE_PRIVATE
+import android.content.SharedPreferences
+import androidx.annotation.VisibleForTesting
+import mozilla.components.support.migration.FennecSettingsMigration.FENIX_PREFS_TELEMETRY_KEY
+import mozilla.components.support.migration.FennecSettingsMigration.FENNEC_PREFS_FHR_KEY
+import mozilla.components.support.migration.FennecSettingsMigration.FENNEC_PREFS_TELEMETRY_KEY
+
+/**
+ * Helper for migrating all common Fennec - Fenix settings.
+ */
+internal object FennecSettingsMigration {
+    private lateinit var fenixPrefs: SharedPreferences
+    private lateinit var fennecAppPrefs: SharedPreferences
+
+    @VisibleForTesting const val FENNEC_APP_SHARED_PREFS_NAME = "GeckoApp"
+    @VisibleForTesting const val FENIX_SHARED_PREFS_NAME = "fenix_preferences"
+    @VisibleForTesting const val FENNEC_PREFS_FHR_KEY = "android.not_a_preference.healthreport.uploadEnabled"
+    @VisibleForTesting const val FENNEC_PREFS_TELEMETRY_KEY = "toolkit.telemetry.enabled"
+    @VisibleForTesting const val FENIX_PREFS_TELEMETRY_KEY = "pref_key_telemetry"
+
+    /**
+     * Migrate all Fennec - Fenix common SharedPreferences.
+     */
+    fun migrateSharedPrefs(context: Context): Result<SettingsMigrationResult> {
+        fennecAppPrefs = context.getSharedPreferences(FENNEC_APP_SHARED_PREFS_NAME, MODE_PRIVATE)
+        fenixPrefs = context.getSharedPreferences(FENIX_SHARED_PREFS_NAME, MODE_PRIVATE)
```

Comment:
That's not the right shared pref. From Fenix's Settings.kt:
```
private val appContext = context.applicationContext

val isTelemetryEnabled by booleanPreference(
        appContext.getPreferenceKey(R.string.pref_key_telemetry),
        default = true
    )
```

So the actual pref ends up in something like `shared_prefs/org.mozilla.fenix.preferences.xml` file, or `org.mozilla.fenix.debug_preferences.xml`, or whatever the app id is. There's _also_ `fenix_preferences.xml` but that's not where this pref is read from.

I think if you just use `context.applicationContext` it should be fine though.

Reply:
Thanks!